File: Homework3/Question1/src/GoldenDartFrog.java
public class GoldenDartFrog extends Animal {
    public GoldenDartFrog() {
        super("gold");
    }

    @Override
    public boolean isEdible() {
        return false;
    }

    public static void testGoldenDartFrog() {
        GoldenDartFrog frog = new GoldenDartFrog();
        //Test GoldenDartFrog
        System.out.println(frog.isEdible() == false);
        System.out.println(frog.getColor() == "gold");
    }
}

File: Homework3/GoldenDartFrog.kt
class GoldenDartFrog : Animal {
    constructor() : super("gold")

    override fun isEdible(): Boolean {
        return false
    }

    companion object {
        fun testGoldenDartFrog() {
            val frog = GoldenDartFrog()
            println(frog.isEdible() == false)
            println(frog.getColor() == "gold")
        }
    }
}